valohai_yaml/objs/pipelines/pipeline_parameter.py

```diff
@@ -0,0 +1,66 @@
+from typing import TYPE_CHECKING, List, Optional
+
+from valohai_yaml.lint import LintResult
+from valohai_yaml.objs.base import Item
+from valohai_yaml.objs.utils import check_type_and_listify
+from valohai_yaml.types import LintContext, SerializedDict
+from valohai_yaml.utils.node_socket_utils import split_socket_str
+
+if TYPE_CHECKING:
+    from valohai_yaml.objs import Config, Pipeline
+
+
+class PipelineParameter(Item):
+    """Represents a parameter definition within a pipeline definition."""
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        targets: List[str],
+        value: Optional[str] = None,
+        default: Optional[str] = None,
+    ) -> None:
+        self.name = name
+        self.targets = check_type_and_listify(targets, str)
+        self.default = default if value is None else value
+
+    @classmethod
+    def parse(cls, data: SerializedDict) -> 'PipelineParameter':
+        data = data.copy()
+        # targets can be a string or a list of strings
+        if 'target' in data:
+            if 'targets' in data:
+                raise TypeError("Pipeline parameter cannot have both: target and targets")
+            data['targets'] = [data.pop('target', ())]
```

Comment:
No need for the default, we definitively know `target` is there.
```suggestion
            data['targets'] = [data.pop('target')]
```